instrumentation/rails/lib/opentelemetry/instrumentation/rails/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rails
+      module Middlewares
+        # TracerMiddleware propagates context and instruments Rails requests
+        # by way of its middleware system
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            extracted_context = OpenTelemetry.propagation.http.extract(env)
+            OpenTelemetry::Context.with_current(extracted_context) do
+              tracer.in_span(env['PATH_INFO'], attributes: request_span_attributes(env: env), kind: :server) do |request_span|
+                @app.call(env).tap do |status, headers, response|
+                  set_attributes_after_request(
+                    request_span,
+                    env: env,
+                    status: status,
+                    headers: headers,
+                    response: response
+                  )
+                end
+              end
+            end
+          ensure
+            OpenTelemetry::Trace.current_span(extracted_context).finish if extracted_context
+          end
+
+          private
+
+          def request_destination_info(env)
+            ::Rails.application.routes.router.recognize(ActionDispatch::Request.new(env)) { |_route, params| return params }
+          end
+
+          def set_attributes_after_request(span, env:, status:, headers:, response:)
+            span.status = OpenTelemetry::Trace::Status.http_to_status(status)
+            span.set_attribute('http.status_code', status)
+            span.set_attribute('http.status_text', ::Rack::Utils::HTTP_STATUS_CODES[status])
+
+            controller = env['action_controller.instance']
+            return unless controller
+
+            span.name = "#{controller.class.name}.#{controller.action_name}"
+            span.set_attribute('rails.controller', controller.class.name)
+            span.set_attribute('rails.action', controller.action_name)
```

Comment:
Although we find these very useful, they aren't part of the semantic conventions.

Reply:
Should these be removed?  I wouldn't expect the specs around conventions to cover every library and framework so for something like this are we able to take some liberties?

Reply:
IDK - in general, I've tried to hew pretty close to the semantic conventions in "official" auto-instrumentation. I'll leave it up to other folks to chime in on this point. Technically, we could extend this in our wrapper gem to add them 🤷 . That said, I wonder if the span name has enough already (e.g. these could be added back in using an attributeprocessor in the collector).

Reply:
This is a good question. I think the main goal with the semantic conventions is so that all instrumentation records a set of common attributes for spans representing common scenarios. I'm not completely sure what the guidance is on collecting more than is required. I can see arguments in both directions.

Reply:
Considering we are leaning towards make the op name a combination of the two, I think we could probably get away with omitting it.